src/WorldFacade.js

```diff
@@ -1,733 +1,809 @@
+      });
+    } else {
+      if (this.config.offscreen) {
+        console.warn(
+          "This browser does not support OffscreenCanvas. Using standard canvas fallback."
+        );
+        this.config.offscreen = false;
+      }
+      // code splitting out WorldOnscreen. It's essentially the same as offscreenCanvas.worker.js but communicates with the main thread differently
+      const WorldOnscreen = await import("./components/world.onscreen").then(
+        (module) => module.default
+      );
+      this.#DiceWorld = new WorldOnscreen({
+        canvas: this.canvas,
+        options: this.config,
+        onInitComplete,
+      });
     }
-		// initialize the AmmoJS physics worker
-		this.#DicePhysics.postMessage({
-			action: "init",
-			width: this.canvas.clientWidth,
-			height: this.canvas.clientHeight,
-			options: this.config
-		})
-	}
-
-	#connectWorld(){
-		const channel = new MessageChannel()
-		
-		// create message channel for the visual world and the physics world to communicate through
-		this.#DiceWorld.connect(channel.port1)
-
-		// create message channel for this WorldFacade class to communicate with physics world
-		this.#DicePhysics.postMessage({
-			action: "connect"
-		},[ channel.port2 ])
-	}
-
-	resizeWorld(){
-		// send resize events to workers - debounced for performance
-		const resizeWorkers = () => {
-			this.#DiceWorld.resize({width: this.canvas.clientWidth, height: this.canvas.clientHeight})
-			this.#DicePhysics.postMessage({action: "resize", width: this.canvas.clientWidth, height: this.canvas.clientHeight});
-		}
-		const debounceResize = debounce(resizeWorkers)
-		window.addEventListener("resize", debounceResize)
-	}
+  }
+
+  // Load the AmmoJS physics world
+  #loadPhysics() {
+    // initialize physics world in which AmmoJS runs
+    this.#DicePhysics = new physicsWorker();
+    // set up a promise to be fulfilled when a message comes back from physics.worker indicating init is complete
+    this.#dicePhysicsPromise = new Promise((resolve, reject) => {
+      this.#dicePhysicsResolve = resolve;
+    });
+    this.#DicePhysics.onmessage = (e) => {
+      switch (e.data.action) {
+        case "init-complete":
+          this.#dicePhysicsResolve(); // fulfill promise so other things can run
+      }
+    };
+    // initialize the AmmoJS physics worker
+    this.#DicePhysics.postMessage({
+      action: "init",
+      width: this.canvas.clientWidth,
+      height: this.canvas.clientHeight,
+      options: this.config,
+    });
+  }
+
+  #connectWorld() {
+    const channel = new MessageChannel();
+
+    // create message channel for the visual world and the physics world to communicate through
+    this.#DiceWorld.connect(channel.port1);
+
+    // create message channel for this WorldFacade class to communicate with physics world
+    this.#DicePhysics.postMessage(
+      {
+        action: "connect",
+      },
+      [channel.port2]
+    );
+  }
+
+  resizeWorld({ width, height }) {
```

Comment:
Allowing this function to take in a width & height. We were getting issue #54 regularly. We fixed it for ourselves by forking this library, implementing this change, and manually controlling resize. 

If width and height are not provided then it will default back to what this function was using so it will maintain backwards compatibility.